core/sub.mk

```diff
@@ -24,15 +24,27 @@ define process_early_ta
 early-ta-$1-uuid := $(firstword $(subst ., ,$(notdir $1)))
 gensrcs-y += early-ta-$1
 produce-early-ta-$1 = early_ta_$$(early-ta-$1-uuid).c
-depends-early-ta-$1 = $1 scripts/ta_bin_to_c.py
-recipe-early-ta-$1 = $(PYTHON3) scripts/ta_bin_to_c.py --compress --ta $1 \
+depends-early-ta-$1 = $1 scripts/ts_bin_to_c.py
+recipe-early-ta-$1 = $(PYTHON3) scripts/ts_bin_to_c.py --compress --ta $1 \
 		--out $(sub-dir-out)/early_ta_$$(early-ta-$1-uuid).c
 endef
 $(foreach f, $(EARLY_TA_PATHS), $(eval $(call process_early_ta,$(f))))
 $(foreach f, $(CFG_IN_TREE_EARLY_TAS), $(eval $(call \
 	process_early_ta,$(out-dir)/ta/$(f).stripped.elf)))
 endif
 
+ifeq ($(CFG_WITH_SP)-$(CFG_EMBEDDED_SP),y-y)
```

Comment:
We do need to check for CFG_WITH_SP.  Only CFG_EMBEDDED_SP is automatically set. CFG_WITH_SP is used to enable/disable SPs

Reply:
Wouldn't it make sense to enable `CFG_WITH_SP` automatically with `EMBEDDED_SP_PATHS=y`?
This would also be in line with how `EARLY_TA_PATHS` or `CFG_IN_TREE_EARLY_TAS` enables `CFG_EARLY_TA`

Reply:
Ok, Changed it

Reply:
Thanks!